src/result_processor.c

```diff
@@ -352,7 +352,8 @@ static int rpsortNext_innerLoop(ResultProcessor *rp, SearchResult *r) {
   }
 
   // if our upstream has finished - just change the state to not accumulating, and yield
-  if (rc == RS_RESULT_EOF) {
+  if (rc == RS_RESULT_EOF ||
+      (rc == RS_RESULT_TIMEDOUT && RSGlobalConfig.timeoutPolicy == TimeoutPolicy_Return)) {
```

Comment:
@MeirShpilraien 
The logic change and in the case of `TimeoutPolicy_Return` we do not return `RS_RESULT_TIMEDOUT` but `RS_RESULT_OK`.
This change answers your question from above.

Reply:
But you assume you always have a sorter that will catch the timeout no? This is not the case with FT.AGGREGATE and I believe its a shared code with FT.AGGREGATE no?